ci/Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+@Library('gpuci_shared_lib') _
+
+pipeline {
+  agent any
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    ansiColor('xterm')
+  }
+  stages {
+    stage("Style Checks") {
+      steps {
+        sh "env"
+      }
+    }
+    stage("C++ Matrix Build") {
+      environment {
+        BUILD_MODE = 'pull-request'
+        PARALLEL_LEVEL = '4'
+        CUDA="11.5"
```

Comment:
The `CUDA` variable should be moved to the shared library.

We should also move the `BUILD_MODE` to a top-level pipeline `environment` section, but we might need to change how it's used in our scripts. We should probably have a call about this.